cypress-io/cypress
# feat: display artifact upload durations in the console

- Closes #28237

### Additional details
Test Replay capture artifacts can be fairly large, so it is important to display how long it takes to upload artifacts to the Cypress cloud when recording a run. Each artifact type (screenshots, video, test replay) now displays how long it took to upload after the upload is complete. If an artifact fails to upload, the duration of the request is displayed as well.

### Steps to test

Record a test run that will produce artifacts that are uploaded to the cloud, and observe the duration in the console output.

### How has the user experience changed?
Before:
<img width="825" alt="Screen Shot 2023-11-02 at 12 36 48 PM" src="https://github.com/cypress-io/cypress/assets/142877/e6617c5e-2135-4f63-9b19-9a13137254aa">

After:
<img width="820" alt="Screen Shot 2023-11-02 at 12 31 40 PM" src="https://github.com/cypress-io/cypress/assets/142877/d2a19096-999f-40c9-9f0c-756927eb71ec">

### PR Tasks


- [NA] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [NA] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
